apps/google-analytics-4/frontend/src/components/main-app/ErrorDisplay/ErrorDisplay.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState, useEffect, useMemo } from 'react';
+import { useSDK } from '@contentful/react-apps-toolkit';
+import { SidebarExtensionSDK } from '@contentful/app-sdk';
+import Note from 'components/common/Note/Note';
+import { ApiErrorType, ERROR_TYPE_MAP, isApiErrorType } from 'apis/apiTypes';
+import {
+  DEFAULT_ERR_MSG,
+  INVALID_ARGUMENT_MSG,
+  PERMISSION_DENIED_MSG,
+} from '../constants/noteMessages';
+import HyperLink from 'components/common/HyperLink/HyperLink';
+
+interface Props {
+  error: Error;
+}
+
+const ErrorDisplay = (props: Props) => {
+  const { error } = props;
+  const [errorDisplay, setErrorDisplay] = useState<string | JSX.Element>('');
```

Comment:
I think storing components in a state to be an anti-pattern and potentially dangerous down the road. Right now it will work but I think this is an important React principle we should follow. Generally I think this errorDisplay (potential rename involved here as well) should be some defined object via `type/interface` and then you render your JSX logic down below in your return based on that errorDisplay object and it's data fields.

Reply:
<img width="627" alt="Screenshot 2023-03-27 at 4 22 12 PM" src="https://user-images.githubusercontent.com/124832189/228080218-5ae60c22-5e5c-4c23-a108-e4ab6fafdfce.png">